pkg/security/activitydump/config/config.go

```diff
@@ -0,0 +1,200 @@
+
+	// ActivityDumpPathMergeEnabled defines if path merge should be enabled
+	ActivityDumpPathMergeEnabled bool
+
+	// ActivityDumpTracedCgroupsCount defines the maximum count of cgroups that should be monitored concurrently. Leave this parameter to 0 to prevent the generation
+	// of activity dumps based on cgroups.
+	ActivityDumpTracedCgroupsCount int
+
+	// ActivityDumpTracedEventTypes defines the list of events that should be captured in an activity dump. Leave this
+	// parameter empty to monitor all event types. If not already present, the `exec` event will automatically be added
+	// to this list.
+	ActivityDumpTracedEventTypes []model.EventType
+
+	// ActivityDumpCgroupDumpTimeout defines the cgroup activity dumps timeout.
+	ActivityDumpCgroupDumpTimeout time.Duration
+
+	// ActivityDumpRateLimiter defines the kernel rate of max events per sec for activity dumps.
+	ActivityDumpRateLimiter int
+
+	// ActivityDumpCgroupWaitListTimeout defines the time to wait before a cgroup can be dumped again.
+	ActivityDumpCgroupWaitListTimeout time.Duration
+
+	// ActivityDumpCgroupDifferentiateArgs defines if system-probe should differentiate process nodes using process
+	// arguments for dumps.
+	ActivityDumpCgroupDifferentiateArgs bool
+
+	// ActivityDumpLocalStorageDirectory defines the output directory for the activity dumps and graphs. Leave
+	// this field empty to prevent writing any output to disk.
+	ActivityDumpLocalStorageDirectory string
+
+	// ActivityDumpLocalStorageFormats defines the formats that should be used to persist the activity dumps locally.
+	ActivityDumpLocalStorageFormats []StorageFormat
+
+	// ActivityDumpLocalStorageCompression defines if the local storage should compress the persisted data.
+	ActivityDumpLocalStorageCompression bool
+
+	// ActivityDumpLocalStorageMaxDumpsCount defines the maximum count of activity dumps that should be kept locally.
+	// When the limit is reached, the oldest dumps will be deleted first.
+	ActivityDumpLocalStorageMaxDumpsCount int
+
+	// ActivityDumpRemoteStorageFormats defines the formats that should be used to persist the activity dumps remotely.
+	ActivityDumpRemoteStorageFormats []StorageFormat
+
+	// ActivityDumpSyscallMonitorPeriod defines the minimum amount of time to wait between 2 syscalls event for the same
+	// process.
+	ActivityDumpSyscallMonitorPeriod time.Duration
+
+	// ActivityDumpMaxDumpCountPerWorkload defines the maximum amount of dumps that the agent should send for a workload
+	ActivityDumpMaxDumpCountPerWorkload int
+
+	// # Dynamic configuration fields:
+	// ActivityDumpMaxDumpSize defines the maximum size of a dump
+	ActivityDumpMaxDumpSize func() int
+
+	// ActivityDumpTagRulesEnabled enable the tagging of nodes with matched rules (only for rules having the tag ruleset:thread_score)
+	ActivityDumpTagRulesEnabled bool
+}
+
+func join(pieces ...string) string {
+	return strings.Join(pieces, ".")
+}
+
+func NewConfig() (*Config, error) {
```

Comment:
The main motivators for this were:
1) `runtime_security_config.activity_dump` is already its own configuration namespace
2) the majority of these settings are used by the same object (the activity dump manager)
3) the activity dump manager does not need to know about any config values outside of this scope
4) there are a lot of config values in this namespace
Given all that, I felt that for the sake of modularity splitting this into an independent config object made sense.

Finally - because all these values are used within the probe (as opposed to the cws consumer), if they don't live here they would have to be in the event monitor config. That config is already quite large, and since the cws consumer is the only one uses activity dumps, I didn't feel that these settings belong in there.